tests/glsl-intrinsic/shader-subgroup/shader-subgroup-arithmetic.slang

```diff
@@ -0,0 +1,131 @@
+// not testing glsl or spirv due to crashing issue from some odd code emitting which looks right but fails
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv
+//TEST:SIMPLE(filecheck=CHECK_HLSL): -allow-glsl -stage compute -entry computeMain -target hlsl
+// not testing cuda due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage compute -entry computeMain -target cuda 
+// not testing cpp due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CPP):  -allow-glsl -stage compute -entry computeMain -target cpp
+
+// not testing glsl or spirv due to crashing issue from some odd code emitting which looks right but fails
+// TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+// TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+// #define TEST_when_glsl_subgroupInclusiveXor_and_subgroupInclusiveAdd_are_not_bugged
+
+precision highp float;
+precision highp int;
+
+//TEST_INPUT:ubuffer(data=[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0], stride=4):out,name=outputBuffer
+buffer MyBlockName2
+{
+    uint data[];
+} outputBuffer;
+
+#define local_size_x_v 32
+layout(local_size_x = local_size_x_v) in;
+
+#include "./glsl-test-utils.slang"
+
+void computeMain()
+{
+    // tests are seperate since subset crashes; once this is fixed, make all into 1 
+
+    float test = 1.0f;
+    uint test2 = 1;
+    outputBuffer.data[0] = testFunctionIn1Param(subgroupAdd, local_size_x_v, test);
+    outputBuffer.data[1] = testFunctionIn1Param(subgroupMul, 1, test);
+    outputBuffer.data[2] = testFunctionIn1Param(subgroupMin, test, test);
+    outputBuffer.data[3] = testFunctionIn1Param(subgroupMin, test, test);
+
+    outputBuffer.data[4] = testFunctionInInt1Param(subgroupAnd, 1, test2);
+    outputBuffer.data[5] = testFunctionInInt1Param(subgroupOr, 1, test2);
+    outputBuffer.data[6] = testFunctionInInt1Param(subgroupXor, 0, test2);
+
+//following functions don't have an hlsl implementation
+#ifdef TARGET_HLSL
+    vec[7] = 1;
+    vec[8] = 1;
+    vec[9] = 1;
+    vec[10] = 1;
+    vec[11] = 1;
+    vec[12] = 1;
+    vec[13] = 1;
+    vec[14] = 1;
+#else 
+    //// compare with goal of true with invocation id 3, not 0
+    uint vec[21];
+    vec[7] = testFunctionIn1Param(subgroupExclusiveAdd, 3, test); 
+    vec[8] = testFunctionIn1Param(subgroupExclusiveMul, 1, test); 
+    vec[9] = testFunctionIn1Param(subgroupExclusiveMin, 1, test);
+    vec[10] = testFunctionIn1Param(subgroupExclusiveMax, 1, test);
+    vec[11] = testFunctionInInt1Param(subgroupExclusiveAnd, 1, test2);
+    vec[12] = testFunctionInInt1Param(subgroupExclusiveOr, 1, test2);
+    vec[13] = testFunctionInInt1Param(subgroupExclusiveXor, 1, test2);
```

Comment:
The use of the variable "test" and "test2" is unnecessary here. It makes the reader to scroll up and down, which indicates a room for a mistake and unnecessary complexity. Just use a hard coded value to make it easier to read.
If you are not familiar with writing unit tests, my advice may seem counter productive because it is opposite of the coding style of the regular codes. But the purpose of unit tests is different from the regular programing.